src/engine/plugins/tile.cpp

```diff
@@ -181,20 +287,318 @@ Status TilePlugin::HandleRequest(const api::TileParameters &parameters, std::str
+            //       both these values can be directly looked up for coordinates
+
+            // If forward travel is enabled on this road section, and we haven't seen this
+            // edge-based-node
+            // before
+            if (edge.forward_segment_id.enabled &&
+                edge_based_node_info.count(edge.forward_segment_id.id) == 0)
+            {
+                // Add this edge-based-nodeid as an outgoing from the source intersection
+                auto f = outgoing_edges.find(edge.u);
+                if (f != outgoing_edges.end())
+                {
+                    f->second.push_back(edge.forward_segment_id.id);
+                }
+                else
+                {
+                    outgoing_edges[edge.u] = {edge.forward_segment_id.id};
+                }
+
+                // Add this edge-based-nodeid as an incoming to the target intersection
+                f = incoming_edges.find(edge.v);
+                if (f != incoming_edges.end())
+                {
+                    f->second.push_back(edge.forward_segment_id.id);
+                }
+                else
+                {
+                    incoming_edges[edge.v] = {edge.forward_segment_id.id};
+                }
+
+                edge_based_node_info[edge.forward_segment_id.id] = {
+                    edge.u, edge.v, edge.forward_packed_geometry_id};
+            }
+            // Same as previous block, but everything flipped
+            if (edge.reverse_segment_id.enabled &&
+                edge_based_node_info.count(edge.reverse_segment_id.id) == 0)
+            {
+                auto f = outgoing_edges.find(edge.v);
+                if (f != outgoing_edges.end())
+                {
+                    f->second.push_back(edge.reverse_segment_id.id);
+                }
+                else
+                {
+                    outgoing_edges[edge.v] = {edge.reverse_segment_id.id};
+                }
+
+                f = incoming_edges.find(edge.u);
+                if (f != incoming_edges.end())
+                {
+                    f->second.push_back(edge.reverse_segment_id.id);
+                }
+                else
+                {
+                    incoming_edges[edge.u] = {edge.reverse_segment_id.id};
+                }
+
+                // Save info about this edge-based-node, note reversal from forward
+                // block above.
+                edge_based_node_info[edge.reverse_segment_id.id] = {
+                    edge.v, edge.u, edge.reverse_packed_geometry_id};
+            }
+        }
+
+        // Now, for every edge-based-node that we discovered (edge-based-nodes are sources
+        // and targets of turns).  EBN is short for edge-based-node
+        for (const auto &source_ebn : edge_based_node_info)
+        {
+            // Grab a copy of the geometry leading up to the intersection.
+            std::vector<NodeID> first_geometry;
```

Comment:
This allocates and then deallocates a vector for every eb-node, what about hoisting it outside the loop?